### DISCUSSION TITLE
[9일차-지원] 쿠키와 세션의 차이?

### DISCUSSION THREAD

**Discussion by smarfy99** [zzangCS/zzangInterview#33] [프론트 면접 질문]
쿠키와 세션의 차이와 언제 사용하는지?

**Comment by skyhanull** [comment]
먼저 쿠키란 사용자의 브라우저에 저장되고, 통신할 때 HTTP 헤더에 포함되는 텍스트 데이터 파일입니다.
특정 웹사이트에 접속할 때 발생하며 사용자의 인증을 도와주고 만료기간이 있어 시간이 지나면 자동으로 삭제됩니다.
세션은 클라이언트 정보를 서버에서 저장하는 것을 말합니다. 위와 달리 서버에 저장이 되있어 쿠키보다 보안성이 좋습니다

**Comment by ez1n** [comment]
- **사용하는 이유**
HTTP 프로토콜 환경은 클라이언트가 요청을 한 후 응답을 받으면 연결을 끊어버리거나 통신이 끝나면 상태를 유지하지 않는 특성을 가지기 때문에 서버는 클라이언트가 누구인지 매번 확인해야 한다. 
이러한 HTTP 프로토콜의 약점을 보완하기 위해 사용한다.
    
- **쿠키**
클라이언트(브라우저) 로콜에 저장되는 키와 값이 들어있는 작은 데이터 파일
    
- **세션**
쿠키를 기반하고 있지만, 사용자 정보 파일을 서버 측에서 관리 (서버 자원 활용)

- **차이점**
1. 사용자의 정보가 저장되는 위치 (쿠키 - 브라우저, 세션 - 서버)
2. 세션은 서버의 처리가 필요하기 때문에 요청 속도는 쿠키가 더 빠르다.
3. 세션은 사용자 정보를 서버에서 관리하기 때문에 보안 면에서 우수하다.
4. 쿠키, 세션 모두 만료시간을 설정할 수 있으나 쿠키는 브라우저를 종료해도 만료시간이 지나지 않으면 정보가 계속 남아있을 수 있으나 세션은 브라우저가 종료되면 만료 시간에 상관없이 삭제된다.
